pymongo/cursor.py

```diff
@@ -437,7 +438,7 @@ def close(self) -> None:
 
     def __query_spec(self):
         """Get the spec to use for a query."""
-        operators = {}
+        operators: Dict[str, Any] = {}
```

Comment:
Use `dict` instead of `Dict` due to `Dict` being deprecated in 3.9.

Reply:
`dict` doesn't support generics until 3.9 so we can only use Dict or MutableMapping here.

Reply:
That brings up a good question. Should we support type checking in all versions of python we support. My intuition says yes because a user might want to run mypy with 3.7. I think it would be odd to say you have to run mypy with a different version of python than your app actually uses.

Reply:
Yes, we test with Python 3.7 and 3.10 specifically in GitHub Actions.

Reply:
talked in person, decided we should stick with `dict` for now